pilot/pkg/networking/core/v1alpha3/cluster_builder.go

```diff
@@ -294,6 +294,13 @@ func (cb *ClusterBuilder) applyDestinationRule(mc *MutableCluster, clusterMode C
 
 	if destRule != nil {
 		mc.cluster.Metadata = util.AddConfigInfoMetadata(mc.cluster.Metadata, destRule.Meta)
+
+		// ALPN header rewrite starts mTLS. Skip if TLS mode is SIMPLE.
```

Comment:
I think we probably want MUTUAL as well?

Reply:
FWIW, the original design doc was scoped to SIMPLE. MUTUAL probably makes sense semantically, but what's the existing mechanism for a tlsMode MUTUAL to indicate that it expects a client cert (outside of ALPN)? My concern is that it only works because we don't remove the alpn rewrite today

Reply:
It is part of TLS spec, it sends the "request client cert" bit during the tls handshake.